core/unit_test/TestRange.hpp

```diff
@@ -327,6 +327,63 @@ TEST(TEST_CATEGORY, range_reduce) {
   }
 }
 
+template <typename ExecSpace, typename StaticBatchSize>
+struct TestStaticBatchSize {
+  using view_type = Kokkos::View<int *, ExecSpace>;
+
+  view_type m_flags;
+  view_type result_view;
+
+  struct AtomicAddTag {};
+  struct VerifyAtomicAddTag {};
+
+  size_t N;
+
+  TestStaticBatchSize(const size_t N_)
+      : m_flags(Kokkos::view_alloc(Kokkos::WithoutInitializing, "flags"), N_),
+        result_view(Kokkos::view_alloc(Kokkos::WithoutInitializing, "results"),
+                    N_),
+        N(N_) {}
+
+  void test_batch_size() {
+    Kokkos::deep_copy(m_flags, 0);
+
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<ExecSpace, AtomicAddTag, StaticBatchSize>(0, N),
+        *this);
+
+    // Verify that each flag has been incremented exactly once
+    bool success = true;
+    Kokkos::parallel_reduce(
+        Kokkos::RangePolicy<ExecSpace>(0, N),
+        KOKKOS_CLASS_LAMBDA(const int i, bool &local_success) {
+          if (m_flags(i) != 1) {
+            Kokkos::printf(
+                "TestStaticBatchSize {::test_batch_size_error at %d != %d\n", i,
+                m_flags(i));
+          }
+          local_success = local_success && (m_flags(i) == 1);
+        },
+        Kokkos::LAnd<bool>(success));
+
+    ASSERT_TRUE(success);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const AtomicAddTag, const int i) const {
+    Kokkos::atomic_add(&m_flags(i), 1);
+  }
```

Comment:
Once you get rid of `VerifyAtomicAddTag`, you can get rid of the tag dispatch.
```suggestion
  KOKKOS_INLINE_FUNCTION
  void operator()(const int i) const {
    Kokkos::atomic_add(&m_flags(i), 1);
  }
```

Reply:
I moved the lambda functor into the tagged operator to maintain consistency with other tests in this suite. Please let me know if you'd prefer me not to use tagged dispatch.

Reply:
I cannot strongly opposed to keep consistency 😉